File: frontendAngular/medical/k-medical/src/app/monClass/utilisateur.ts
import { RoleUtilisateur } from "./Roles";
import { Specialite } from "./Specialitet";

 
export class Utilisateur {
	    id:number;
	    nom:string;
	    prenom:string;
	    adresse:string;
	    tel:string;
 	    cin:string;
	    email:string;
	    password:string;
	    etat:boolean;
	    dateNaissance:string;
		confirmPasswrod:string
  	    roleUtilisateurs:RoleUtilisateur[];
  
}

File: backendSpringboot/k-medical/src/main/java/com/medical/model/Utilisateur.java
package com.medical.model;

import java.util.Set;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Inheritance;
import jakarta.persistence.InheritanceType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Inheritance(strategy = InheritanceType.JOINED)  

public class Utilisateur {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private String nom;
	private String prenom;
	private String adresse;
	private String tel;
	private String cin;
    private String resetToken;

 	private String email;
	private String password;	
	private String confirmPasswrod;	
	private String dateNaissance;
	private boolean etat;

	@ManyToMany(fetch = FetchType.EAGER, cascade = CascadeType.MERGE)
	@JoinTable(name = "UTILISATEUR_ROLE", joinColumns = { @JoinColumn(name = "ID_UTILISATEUR") }, inverseJoinColumns = {
			@JoinColumn(name = "ID_ROLE") })
	private Set<RoleUtilisateur> roleUtilisateurs;
}
